msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/AadInstanceDiscoveryTest.java

```diff
@@ -186,4 +188,137 @@ public void aadInstanceDiscoveryTest_AutoDetectRegion_NoRegionDetected() throws
         Assert.assertTrue(entry.aliases().contains("login.microsoft.com"));
         Assert.assertTrue(entry.aliases().contains("sts.windows.net"));
     }
+
+    @DataProvider(name = "aadClouds")
+    private static Object[][] getAadClouds(){
+        return new Object[][] {{"https://login.microsoftonline.com/common"} , // #Known to Microsoft
+                {"https://private.cloud/foo"}//Private Cloud
```

Comment:
Consider adding some of the sovereign clouds here

Reply:
Would the instance discovery behavior for sovereign clouds be different in some way? Is it worth having both global and sovereign cloud tests here?

Reply:
Well, I don't know your codebase, but there may special rules related to public cloud vs sovereign cloud. For example, ESTS-R feature handles these 2 differently and it is also in this general area of the code. So when thinking about different clouds, it's good to have a test case for a sovereign cloud.